teps/000X-trigger-crd.md

```diff
@@ -0,0 +1,366 @@
+```
+
+#### Trigger Status
+Trigger status will report url of EventListener bound and EventListener information to 
+which it is bound.
+
+### EventListener CRD
+EventListeners expose an addressable "Sink" to which incoming events in the form of HTTP requests
+are directed and process those requests with configuration from bound Triggers. Boundnamespace acts 
+as selector based on namespace. All triggers in namespaces specified are selected can be served by
+EventListener.
+
+```
+apiVersion: v1alpha1
+kind: EventListener
+metadata:
+  name: my-repo-trigger
+spec:
+  serviceAccountName: "blah"
+  boundnamespace:
+      - foo-ns
+      - bar-ns
+```
+
+### Path based EventListener
+Inside EventListener, we will deduce Trigger to execute based on the path of the request URL.
+To refer to trigger resource foo in namespace bar, we would url ```/bar/foo```. Instead of 
+EventListener iterating through every Triggers that can be serve by EventListener, EventListener
+will directly process the Trigger refer by URL.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this TEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+We have two type of users:
+1. End users who requires his event to be processed.
+2. Operator/admin who managed EventListener.
+#### Trigger Ref inside the EventListener
+Instead of defining Trigger inside EventListener, Trigger resource will be created by end user.
+Trigger will contain serviceaccount, triggerbinding, triggertemplate and interceptor.ServiceAccount 
+will be optional. If not defined, ServiceAccount of EventListener will be used.
+
+#### Specifying Namespaces where EventListener can serve
+Admin/Operator would give which namespaces EventListener can serve. In this way, EventListener can be used
+in Multitenant Scenarios.
+
+#### Path Base EventListener Implementation
+Path base EventListener will be implemented.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+1. Security issue - Wide permission to EventListener. In this scenario EventListener
+will function in the same way as controller. It will be the responsibility of Operator
+or admin to manage that.
+2. Log: How will user access the EventListener log to debug their events.
```

Comment:
https://github.com/tektoncd/cli/issues/1030 could be useful here
as well as https://github.com/tektoncd/triggers/issues/649

Reply:
Added those.